test/acceptance/framework/consul/consul_cluster.go

```diff
@@ -287,6 +287,19 @@ func (h *HelmCluster) checkForPriorInstallations(t *testing.T) {
 	for _, r := range installedReleases {
 		require.NotContains(t, r["chart"], "consul", fmt.Sprintf("detected an existing installation of Consul %s, release name: %s", r["chart"], r["name"]))
 	}
+
+	// Wait for all pods labelled with app=consul to exit. A previous
+	// release may not be listed by Helm but its pods may still be terminating.
+	retry.RunWith(&retry.Counter{Wait: 1 * time.Second, Count: 30}, t, func(r *retry.R) {
+		consulPods, err := h.kubernetesClient.CoreV1().Pods(h.helmOptions.KubectlOptions.Namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=consul"})
+		require.NoError(r, err)
+		var podNames []string
+		for _, p := range consulPods.Items {
+			podNames = append(podNames, p.Name)
+		}
+		require.Len(r, podNames, 0)
```

Comment:
Any reason we can't do a `require.Len(r, consulPods.Items, 0)`?

Reply:
Oops, I thought `require.Len` was the one that prints out the contents of each slice but that's actually `require.ElementsMatch` I think. Basically I want it to print out the pods that are still there so I think this should be

```go
		consulPods, err := h.kubernetesClient.CoreV1().Pods(h.helmOptions.KubectlOptions.Namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=consul"})
		require.NoError(r, err)
		if len(consulPods.Items) > 0 {
			var podNames []string
			for _, p := range consulPods.Items {
				podNames = append(podNames, p.Name)
			}
			r.Errorf("pods from previous installation still running: %s", strings.Join(podNames, ", "))
		}

```

Reply:
heh. makes sense! you're right about `ElementsMatch` thought i suspect that would be `ElemsMatch(r, podNames, [])` but the rationale makes sense!!

Reply:
Right but in that case the error will print out the go-formatted entire pod object so this way it only prints out the pod names.

Reply:
you have a beautiful 🧠 @lkysow